stock_intercompany/models/stock_picking.py

```diff
@@ -12,78 +12,99 @@ def _create_counterpart_picking(self):
         picking = self.env["stock.picking"]
         if self.partner_id in partners:
             company = partners[self.partner_id]
+            # Switch to target company context before creating picking
+            picking = self.env["stock.picking"].sudo().with_company(company)
```

Comment:
```suggestion
            picking_model = self.env["stock.picking"].sudo().with_company(company)
```
Or any other name, just to not confuse with the picking you create right after